### PR TITLE
mgr/orchestrator: Simplify Orchestrator wait implementation 

### PR DESCRIPTION
**Pull request by jmolmo**
Avoid to obtain the loop condition value inside the loop. This makes possible to remove the redundant comparation with an empty list in the assignation of the done variable.

Without the redundant comparation the code was stopped in the moment of the boolean evaluation of the completion list.

Signed-off-by: Juan Miguel Olmo Martínez <jolmomar@redhat.com>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
Could you amend your PR title and commit message title to be a bit more specific?

File: src/pybind/mgr/orchestrator_cli/module.py [hunk 1]
```diff
@@ -82,27 +82,24 @@ def _wait(self, completions):
 
         Waits for writes to be *persistent* but not *effective*.
         """
-        done = False
-
-        while done is False:
-            done = self._oremote("wait", completions) == []
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator_cli/module.py]: Ok, after looking at the code, `self._oremote("wait", completions) == []` is always False, because `wait()` returns a boolean, not a list.

File: src/pybind/mgr/orchestrator_cli/module.py [hunk 2]
```diff
@@ -82,27 +82,24 @@ def _wait(self, completions):
...
-                            any_nonpersistent = True
-                            break
-
-                if any_nonpersistent:
-                    time.sleep(5)
+
+        while self._oremote("wait", completions):
+
+            any_nonpersistent = False
+            for c in completions:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator_cli/module.py]: what about changing this to 
```python
any_nonpersistent = any((c.is_read and not c.is_compete) or (not c.is_read and not c.is_persistent) for c in completions)
# afaict same as
any_nonpersistent = any(not (not c.is_read or c.is_compete) or not (c.is_read or c.is_persistent) for c in completions)
# afaict same as
any_nonpersistent = any(not ((not c.is_read or c.is_compete) and (c.is_read or c.is_persistent)) for c in completions)
```
?

File: src/pybind/mgr/orchestrator_cli/module.py (hunk 1 above)
Comment (jmolmo) [reply src/pybind/mgr/orchestrator_cli/module.py]: Not in the case of the Ansible Orchestrator, that returned the list of completions objects with operations not finished or an empty list if all operations finish.
As you say, and it is stated in the documentation, the orchestrator wait method MUST return a boolean. So i have fixed this in the ansible orchestrator.
However, this does not explain the weird behavior when a returned empty list is deemed "not False".

File: src/pybind/mgr/orchestrator_cli/module.py (hunk 2 above)
Comment (jmolmo) [reply src/pybind/mgr/orchestrator_cli/module.py]: Brilliant idea!! I have added the "should_wait" attribute to the completion objects. It simplifies and make more easy to understand the behavior in the "wait" calls/workflow.

**Comment by tserong** [comment]
Yeah, that's a much simplified implementation. Nice work.

**Comment by jmolmo** [comment]
Squashed and amended commit message. Also "Rebased" but the unit test error in "librbd.dir/all" persist.( i think not related at all with this PR)

File: src/pybind/mgr/orchestrator_cli/module.py [hunk 3]
```diff
@@ -124,7 +109,7 @@ def _list_devices(self, cmd):
         >
         > or human readable:
         >
-        >     HOST  DEV  SIZE  DEVID(vendor\_model\_serial)   IN-USE  TIMESTAMP
+        >     HOST  DEV  SIZE  DEVID(vendor\\_model\\_serial)   IN-USE  TIMESTAMP
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator_cli/module.py]: Just curiosity: why did you need this additional `\`?
